### PR TITLE
XW-3981 | prevent duplicated LyricFind calls

### PR DESCRIPTION
**Pull request by xkxd**
## Links

https://wikia-inc.atlassian.net/browse/XW-3981

## Description

LyricFind tracking requests are duplicated, so we're trying to prevent that

## Reviewers

@Wikia/x-wing 
@frankfarmer - Request will be sent only by our Fastboot service for server-side rendering, I've also added "post-sent"  log with all request headers included - will that suit your needs?

### CODE REVIEW

File: app/routes/wiki-page.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ export default Route.extend(
 		},
 
 		/**
-		 * @param {Ember.Object} model
+		 * @param {Ember.model} model
```
Comment (latata) [inline app/routes/wiki-page.js]: What's `Ember.model`?

**Review by latata — APPROVED** [review]

File: app/utils/wiki-handlers/article.js [hunk 2]
```diff
@@ -34,31 +34,46 @@ function afterModel(route, model) {
...
+ * @param {number} wikiId
+ * @param {Ember.Object} headers - FastBoot request's headers
+ * @returns {boolean}
+ */
+function shouldSendLyricFindRequest({model, wikiId, headers}) {
+	const lyricWikiId = 43339;
+
+	return wikiId === lyricWikiId
+		&& !model.get("isMainPage")
+		&& headers.get("X-Wikia-Is-Internal-Request") !== "goreplay";
```
Comment (rogatty) [inline app/utils/wiki-handlers/article.js]: Please add a comment explaining what is `goreplay`. And use single quotes everywhere ;)

**Review by rogatty — APPROVED** [review]

File: app/utils/wiki-handlers/article.js [hunk 3]
```diff
@@ -34,42 +34,63 @@ function afterModel(route, model) {
...
+ * @returns {boolean}
+ */
+function shouldSendLyricFindRequest({model, wikiId, headers}) {
+	const lyricWikiId = 43339;
+
+	// 'goreplay' header indicates http traffic replayed between SJC and RES
+	// for more info please read 'Confluence > Operations > Goreplay' article
+	return wikiId === lyricWikiId
+		&& !model.get('isMainPage')
+		&& headers.get('X-Wikia-Is-Internal-Request') !== 'goreplay';
```
Comment (frankfarmer) [inline app/utils/wiki-handlers/article.js:78]: Honestly, the presence of the `X-Wikia-Is-Internal-Request` header (regardless of value) is probably reason enough not to send a beacon.  But this stop the bulk of the problem for now anyway.

**Comment by frankfarmer** [comment]
> I've also added "post-sent" log with all request headers included

Sounds great.

**Pull request merged**